modules/gapi/src/backends/common/serialization.cpp

```diff
@@ -361,6 +366,140 @@ I::IStream& operator>> (I::IStream& is, cv::GArg &arg) {
     return is;
 }
 
+I::OStream& operator<< (I::OStream& os, const cv::GMetaArg &arg) {
+    os << (uint32_t)arg.index();
+    switch (arg.index()) {
+    case cv::GMetaArg::index_of<util::monostate>() :
+        os << cv::util::get<cv::util::monostate>(arg);       break;
+    case cv::GMetaArg::index_of<cv::GMatDesc>() :
+        os << cv::util::get<cv::GMatDesc>(arg);              break;
+    case cv::GMetaArg::index_of<cv::GScalarDesc>() :
+        os << cv::util::get<cv::GScalarDesc>(arg);           break;
+    case cv::GMetaArg::index_of<cv::GArrayDesc>() :
+        os << cv::util::get<cv::GArrayDesc>(arg);            break;
+    case cv::GMetaArg::index_of<cv::GOpaqueDesc>() :
+        os << cv::util::get<cv::GOpaqueDesc>(arg);           break;
+    default: GAPI_Assert(false && "GMetaArg: Unsupported type for <<");
+    }
+    return os;
+}
```

Comment:
You don't need this function I believe, there is already a generic overload for `util::variant<...>`

Reply:
done